packages/opencensus-core/src/stats/metric-producer.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Metric} from '../metrics/export/types';
+import {Stats} from './stats';
+
+/**
+ * A MetricProducer producer that can be registered for exporting using
+ * MetricProducerManager.
+ */
+export class MetricProducer {
```

Comment:
At high level, StatsManager is one of the MetricProducer, (another is MetricRegisty: gauges). I think you should add seperate abstract class or interface for MetricProducer.
example:
```
export abstract class MetricProducer {
     getMetrics(): Metric[];
}
```

and use that here. Something like.

```
export class MetricProducerForStats extends MetricProducer {
     // ...
}
```

Reply:
Done. Thank you